concent_api/core/message_handlers.py

```diff
@@ -519,18 +532,42 @@ def handle_send_force_subtask_results(client_message: message.concents.ForceSubt
             reason=message.concents.ForceSubtaskResultsRejected.REASON.RequestPremature,
         )
 
-    subtask = store_or_update_subtask(
-        task_id                     = task_to_compute.compute_task_def['task_id'],
-        subtask_id                  = task_to_compute.compute_task_def['subtask_id'],
-        provider_public_key         = provider_public_key,
-        requestor_public_key        = requestor_public_key,
-        state                       = Subtask.SubtaskState.FORCING_ACCEPTANCE,
-        next_deadline               = forcing_acceptance_deadline + settings.CONCENT_MESSAGING_TIME,
-        set_next_deadline           = True,
-        ack_report_computed_task    = client_message.ack_report_computed_task,
-        task_to_compute             = client_message.ack_report_computed_task.report_computed_task.task_to_compute,
-        report_computed_task        = client_message.ack_report_computed_task.report_computed_task,
+    subtask = get_one_or_none(
+        Subtask,
+        subtask_id=task_to_compute.compute_task_def['subtask_id'],
     )
+
+    if subtask is None:
+        subtask = store_subtask(
+            task_id=task_to_compute.compute_task_def['task_id'],
+            subtask_id=task_to_compute.compute_task_def['subtask_id'],
+            provider_public_key=provider_public_key,
+            requestor_public_key=requestor_public_key,
+            state=Subtask.SubtaskState.FORCING_ACCEPTANCE,
+            next_deadline=forcing_acceptance_deadline + settings.CONCENT_MESSAGING_TIME,
+            task_to_compute=client_message.ack_report_computed_task.report_computed_task.task_to_compute,
+            report_computed_task=client_message.ack_report_computed_task.report_computed_task,
+            ack_report_computed_task=client_message.ack_report_computed_task,
+        )
+    else:
+        if subtask.state_enum == Subtask.SubtaskState.FORCING_ACCEPTANCE:
+            return message.concents.ServiceRefused(
+                reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+            )
+        if task_to_compute is not None and subtask.task_to_compute is not None:
```

Comment:
Same as above. This should be an `assert`.

Reply:
the same situation as above: 
` if subtask is None:`
`.....`
`else:`
`... <- I think it is unnecessary to assert if subtask is not none in this place. It is obvious..`